src/services/connection.service.ts

```diff
@@ -23,6 +24,30 @@ class ConnectionService {
     });
     return userConnections;
   }
+  public async sendConnectionRequest(data: SendOrAcceptConnectionRequestDto): Promise<BaseConnectionsDto> {
+    const { senderId, receiverId } = data;
+    const connectionRequest = await ConnectionsEntity.find({
```

Comment:
we can also just use findOne, if there is at most 1 connection between the sender and receiver. This way you don't have a list that, at max is of length 1. This way you dont need to check if the length is > 0, you can just check if the return from the await exists. Unless you plan on saving rejected connections.

Reply:
Going to update it to use findOne